src/Makefile.am

```diff
@@ -25,21 +29,43 @@ sources += platform-aws.c
 endif
 
 if ENABLE_NEURON
-  lib_LTLIBRARIES = libnccom-net.la
   sources += nccl_ofi_interface_neuron.c
-  libnccom_net_la_SOURCES = $(sources)
 else
-  lib_LTLIBRARIES = libnccl-net.la
   sources += nccl_ofi_cuda.c \
 	nccl_ofi_interface_nvidia.c
-  libnccl_net_la_SOURCES = $(sources)
+endif
+
+# Build an internal-only library that can be used by unit tests as
+# well as the actual nccl_net.so / nccom_net.so libraries.  This saves
+# us writing dlopen() handlers for simple unit tests.
+noinst_LTLIBRARIES = internal_net_plugin.la
+internal_net_plugin_la_SOURCES = $(sources)
+internal_net_plugin_la_LDFLAGS = -module -avoid-version
```

Comment:
I'm probably missing something here... if `internal_net_plugin.la` and `libnccl-net.la` are compiled with the same flags (`-module -avoid-version`), then why do we need both?